protostar/commands/test/starkware/test_execution_state.py

```diff
@@ -36,7 +38,15 @@ async def from_test_suite_definition(
             contract=contract,
             starknet_compiler=starknet_compiler,
             context=TestContext(),
+            output_recorder=OutputRecorder(),
         )
 
+    def get_output(self, name: OutputName) -> str:
+        return self.output_recorder.captures[name].getvalue()
+
     def fork(self) -> Self:
-        return dataclasses.replace(super().fork(), context=deepcopy(self.context))
+        return dataclasses.replace(
+            super().fork(),
+            context=deepcopy(self.context),
+            output_recorder=deepcopy(self.output_recorder),
```

Comment:
seems like encapsulation here could be nice:

```suggestion
            output_recorder=self.output_recorder.fork(),
```